### ISSUE TITLE
Lifecycle Manager

### ISSUE THREAD

**Issue by pihme** [camunda-cloud/zeebe#7538]
**Description**

Create a class to track the status of a startup shutdown process. More precisely, it should track the status according to the following state diagram:

![image](https://user-images.githubusercontent.com/14032870/126527177-14532ed4-4380-41d5-8c00-c431af9d71da.png)

***Signature***

```
public class LifecycleManager {
  LifecycleManager(StartupProcess startupProcess);
  ActorFuture<CONTEXT> startup(final CONTEXT context);
  ActorFuture<CONTEXT> shutdown(final CONTEXT context); 
  Status getStatus();
  boolean isStarting() isStarted() isShuttingDown() isShutdown();
}
```

Maybe we also want to add lifecycle listeners in the future, or connect to monitoring in some way. But for a start, all it should do is keep track of the state related to a startup and shutdown process.

**Comment by pihme** [comment]
Depends on #7536

**Comment by npepinpe** [comment]
What is the difference between this and an actor's lifecycle? The actor has a few more states, but otherwise can be mapped to this. So let me rephrase, what's the difference between this and using, say, a root actor?

**Comment by pihme** [comment]
Good question.

Here is my problem with the actor lifecycle as it is implemented today: It changes when "there is nothing else to do". 

```
private boolean onAllJobsDone() {
    boolean resubmit = false;

    if (allPhaseSubscriptionsTriggered()) {
      switch (lifecyclePhase) {
        case STARTING:
          lifecyclePhase = ActorLifecyclePhase.STARTED;
          submitStartedJob();
          startingFuture.completeWith(jobStartingTaskFuture);
          resubmit = true;
          break;
```

(maybe my problem is also that I don't groc the actor classes)

What I would prefer is some way to explicitly switch the state. So something like:
```
CompletableFuture<Void> onActorStarting() {
  final var result = new CompletableFuture();
  actor.submit( {} -> httpclient.sendRequest(myRequest, 
                                                    (response) -> {
                                                        actor.submit({} -> {
                                                           handleResponse();
                                                           result.complete();
                                                       });
                                                  });
  return result;
}
```
And then the completion of the future triggers the state change.

The way it is implemented now I feel I have little control over when the state change happens. All I can do is schedule tasks. So in the scenario above the state change might happen after the request is sent, but before the response is received.

Or I could control it by submitting a task that blocks the thread until the response is received. But blocking the thread seems to be against the idea of using actors in the first place.

So you see why I avoid the `Actor` classes, because they weird me out - or I just don't understand how to use them.

But I agree it would be better to reuse and improve on existing abstractions rather than introducing new one.

At the same time, I didn't plan on changing the lifecycle of actors in any drastic way. Because then I would have to touch all actors.

So let's think a little harder how we want to tackle this part.

**Comment by pihme** [comment]
This issue is a YAGNI candidate as long as we have no component which uses the status.
